R/decisiontreeutils.R

```diff
@@ -0,0 +1,431 @@
+    if(max_bins < 2) {
+      stop("The minimum bins is 2")
+    } else {
+      param_list$maxNumBins <- max_bins
+    }
+  }
+
+  # other parameters
+  param_list$minsplit <- config$min.split
+  param_list$minbucket <- config$min.bucket
+  param_list$xval <- config$xval.folds
+  param_list$maxdepth <- config$max.depth
+
+  param_list$cp <- ifelse(config$the.cp == "Auto" || config$the.cp == "", .00001, config$the.cp)
+
+  param_list
+}
+
+#' name mapping from parameters to functions
+#'
+#' @param f_string string of function
+#' @param param_list list of decision tree params
+#' @return list with named parameters for f_string
+paramsToDTArgs <- function(f_string, param_list) {
+  if (f_string == "rpart") {
+    list(
+      data = param_list$data,
+      formula = param_list$f,
+      weights = param_list$weight_arg,
+      method = param_list$method,
+      parms = param_list$parms,
+      usesurrogate = param_list$surrogate,
+      minsplit <- param_list$minsplit,
+      minbucket = param_list$minbucket,
+      xval = param_list$xval,
+      maxdepth = param_list$maxdepth,
+      cp = param_list$cp
+    )
+  } else if(f_string == "rxDTree") {
+    list(
+      data = quote(xdf_path),
+      formula = param_list$f,
+      pweights = param_list$weight_arg,
+      method = param_list$method,
+      parms = param_list$parms,
+      useSurrogate = param_list$surrogate,
+      maxNumBins = param_list$max_bins,
+      minSplit <- param_list$minsplit,
+      minBucket = param_list$minbucket,
+      xVal = param_list$xval,
+      maxDepth = param_list$maxdepth,
+      cp = param_list$cp
+    )
+  } else {
+    stop(paste("Unsupported function specified: ", f_string))
+  }
+}
+
+#' adjusts config based on results if config was initially "Auto"
+#'
+#' @param config list of config options
+#' @param the_model model object
+#' @return model obj after adjusting complexity parameter
+adjustCP <- function(config, the_model) {
+  if(is.na(as.numeric(config$the.cp)) && (config$the.cp == "Auto" || config$the.cp == "")) {
+    cp_table <- as.data.frame(the_model$cptable)
+    pos_cp <- cp_table$CP[(cp_table$xerror - 0.5*cp_table$xstd) <= min(cp_table$xerror)]
+    new_cp <- pos_cp[1]
+    print(cp_table)
+    if (cp_table$xerror[1] == min(cp_table$xerror)) {
+      stop.Alteryx("The minimum cross validation error occurs for a CP value where there are no splits. Specify a complexity parameter and try again.")
+    }
+    prune(the_model, cp = new_cp)
+  } else {
+    the_model
+  }
+}
+
+#' get grp|out pipes for outputting static report
+#'
+#' @param config list of config options
+#' @param the_model model object
+#' @param is_XDF boolean of whether model is XDF
+#' @return dataframe of piped results
+getDTPipes <- function(config, the_model, is_XDF) {
+
+  # The output: Start with the pruning table (have rxDTree objects add rpart
+  # inheritance for printing and plotting purposes).
+  if (is_XDF) {
+    the_model_rpart <- rxAddInheritance(the_model)
```

Comment:
From a testing perspective, I would refactor the logic for XDF output into a separate function, that can then get called inside `getDTPipes`.